### DISCUSSION TITLE
Possible to use SQLite?

### DISCUSSION THREAD

**Discussion by unleashit** [Infisical/infisical#88] [Q&A] [answered]
Apologies for not having a chance to dig too deeply yet, but is it possible to use SQlite as a database, or does the self hosted version require Mongo? This looks like a great app, but the extra overhead for smaller projects seems unnecessary.

I'm just going by what I saw in the docs for self hosting and docker. 

Also wondering if for some reason there is a particular dependency on nginx (or it's image) since that was included in the setup. I'm sure it's just trying to be helpful to people who aren't used to self hosting, but it's usually left out since most people are either getting started by testing locally (where you usually don't need a proxy) or chances are you might already have one set up.

**Comment by dangtony98** [comment, marked as answer]
Hey @unleashit!

Infisical uses MongoDB as its database, so the self-hosted version would require it; understandably, it might be a lot of overhead for smaller projects — would be cool to have an Infisical-mini one day haha.

Regarding nginx, we include a basic nginx setup for people to get up and running; it routes requests to either the frontend or backend (this is an unfortunate result of one of our early design decisions to split the frontend and backend into separate images; in the future ideally you won't even need a reverse-proxy to run Infisical when it's all squeezed into 1 image). 

In practice, you can configure and use your own reverse-proxy for Infisical such as traefik which some users have done but yea everything is flexible so consider nginx as the boiilerplate setup.

Feel free to join our [Slack](https://infisical-users.slack.com/ssb/redirect) community to chat about it :)

**Comment by unleashit** [comment]
Hey and thanks for the quick reply. It's all good and obviously it's a pretty young project. What you have so far looks pretty interesting, and easier compared to something like Vault. Still I think at least for my hobby projects, I'll wait for the Infiscal-mini :p Maybe you can go with an ORM like Prisma to make it easier?

Gotcha as far as nginx, but couldn't that be fixed with a CORS middleware on the backend?

**Reply by dangtony98** [reply]
Yeah that's what we're trying to do! — Create a more intuitive and simpler secret manager that'll come with a lot of modern features in the future. I'll take you up on the Infisical-Mini but probably won't be sometime till we get this Infisical down first haha.

We could consider the ORM like Prisma but the work done so far has been built around Mongoose so it might be a lot to shuffle around given many ongoing developments in the codebase.

Regarding nginx, not sure. Either way, I think we'll be squeezing everything into 1 image anyways so it'll be removed.

**Reply by unleashit** [reply]
Going off the video, it has a great UX. Starred and will be watching! I mentioned Prisma because it supports Mongo, but I realize it isn't trivial to switch.
